src/test/scala/mesosphere/marathon/health/MarathonHealthCheckManagerTest.scala

```diff
@@ -306,38 +305,5 @@ class MarathonHealthCheckManagerTest extends MarathonSpec with Logging {
     assert(hcManager.list(otherAppId) == otherHealthChecks)
   }
 
-  /**
-    * Captures the events send to the EventStream while the block is executing.
-    *
-    * For issue #2314 not only the end state is important. It is also important, which health checks
-    * were removed incorrectly and then readded afterwards.
-    */
-  private[this] def captureEvents(block: => Unit): Seq[MarathonEvent] = {
-    // yes, this is ugly. Since we only access it in the actor until it terminates, we do have
-    // the correct thread sync boundaries in place.
-
-    var capture = Vector.empty[MarathonEvent]
-    val captureEventsActor = actor {
-      new Act {
-        become {
-          case captureMe: MarathonEvent => capture :+= captureMe
-        }
-      }
-    }
-    eventStream.subscribe(captureEventsActor, classOf[MarathonEvent])
-    eventStream.subscribe(captureEventsActor, classOf[String])
-
-    try {
-      block
-    }
-    finally {
-      eventStream.unsubscribe(captureEventsActor)
-      captureEventsActor ! PoisonPill
-      val probe = TestProbe()
-      probe.watch(captureEventsActor)
-      probe.expectMsgClass(classOf[Terminated])
-    }
-
-    capture
-  }
+  def captureEvents = CaptureEvents(eventStream)
```

Comment:
No, performance was not bothering me here at all. I completely understand that you like the syntactic sugar a case class comes with; I'd just prefer having a concise syntax where I can see the type of what I'm supposed to use. The way you implemented it is nice and dotless – however, one has to _know_ that `CaptureEvents` is a case class, and that a reference to it can therefore be used to _implicitly_ call the apply method. I can imagine thinking differently about that tomorrow; let's just leave it like that.